hibernate-core/src/main/java/org/hibernate/hql/spi/id/inline/InlineIdsInClauseBuilder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.hql.spi.id.inline;
+
+import java.util.List;
+
+/**
+ * Builds the where IN clause that wraps the identifiers to be updated/deleted.
+ *
+ * @author Vlad Mihalcea
+ */
+public class InlineIdsInClauseBuilder extends IdsClauseBuilder {
+
+	private final int chunkLimit;
+
+	private final StringBuilder statementBuilder;
+
+	public InlineIdsInClauseBuilder(
+			String[] columns, List<Object[]> ids, int chunkLimit) {
+		super( columns, ids );
+		this.chunkLimit = chunkLimit;
+		this.statementBuilder = buildStatement();
+	}
+
+	public String toStatement() {
+		return statementBuilder.toString();
+	}
+
+	private StringBuilder buildStatement() {
+		StringBuilder buffer = new StringBuilder();
+
+		String columnNames = String.join( ",", (CharSequence[]) columns() );
+
+		for ( int i = 0; i < ids().size(); i++ ) {
+			Object[] idTokens = ids().get( i );
+			if ( i > 0 ) {
+				if( chunkLimit > 0 && i % chunkLimit == 0 ) {
+					buffer.append( " ) or ( " );
+					buffer.append( columnNames );
+					buffer.append( " ) in (" );
+				}
+				else {
+					buffer.append( "," );
+				}
+			}
+			buffer.append( "(" );
+			for ( int j = 0; j < idTokens.length; j++ ) {
+				if ( j > 0 ) {
+					buffer.append( "," );
+				}
+				buffer.append( idTokens[j] );
+			}
+			buffer.append( ")" );
+		}
+
+		return buffer;
```

Comment:
It could be interesting, but I like that the flow it's much easier to follow now (being more readable) even at the price of a tiny code duplication. I'll leave it as is.